tools/resourcedocsgen/cmd/metadata.go

```diff
@@ -376,6 +382,27 @@ func legacyNameRuleException(schemaName, providerName string) bool {
 	return schemaName == "runpod" && providerName == "runpod-native"
 }
 
+// legacyPublisherRuleException checks if the repository slug is an exception to the
+// standard publisher naming rules. This allows certain repositories to maintain
+// backward compatibility with the previous behavior of not requiring a publisher field
+// in the schema.
+//
+// For existing providers, we allow the publisher field to be empty and derive it from
+// the repository owner. New providers should not be added to this exception list and
+// should explicitly specify their publisher in the schema.
+func legacyPublisherRuleException(repoSlug repoSlug) bool {
+	return repoSlug.String() == "genesiscloud/pulumi-genesiscloud"
+}
+
+// getLegacyPublisher returns the publisher name for repositories that are exceptions
+// to the standard publisher naming rules. It converts the repository owner to title case
+// to derive a publisher name when one is not explicitly specified in the schema.
+// This maintains backward compatibility with previously established publisher names.
+func getLegacyPublisher(repoSlug repoSlug) string {
+	contract.Assertf(repoSlug.owner != "", "repoSlug.owner is non-empty by construction")
+	return cases.Title(language.Und, cases.NoLower).String(repoSlug.owner)
```

Comment:
Here you can see how we constructed the name previously: https://github.com/pulumi/registry/pull/6246/files#diff-065252e9bd0b09c1de98e6a444c4ddc189efff4334c66b19e2ff75a1970f29ecL188-L189